### PR TITLE
chore: Integ tests for fail-on-empty-changeset flag

### PR DESCRIPTION
**Pull request by jfuss**
*Issue #, if available:*

*Description of changes:*

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration/deploy/test_deploy_command.py [hunk 1]
```diff
@@ -344,6 +344,110 @@ def test_deploy_with_s3_bucket_switch_region(self, template_file):
...
+        deploy_process_execute = Popen(deploy_command_list, stdout=PIPE)
+        try:
+            deploy_process_execute.communicate(timeout=TIMEOUT)
+        except TimeoutExpired:
+            deploy_process_execute.kill()
+            raise
+        # Deploy should succeed
+        self.assertEqual(deploy_process_execute.returncode, 0)
+
+        # Try to deploy to another region.
```
Comment (TheSriram) [inline tests/integration/deploy/test_deploy_command.py]: we are deploying to the same region here.

File: tests/integration/deploy/test_deploy_command.py [hunk 2]
```diff
@@ -344,6 +344,110 @@ def test_deploy_with_s3_bucket_switch_region(self, template_file):
...
+        deploy_command_list = self.get_deploy_command_list(fail_on_empty_changeset=True, **kwargs)
+
+        deploy_process_execute = Popen(deploy_command_list, stdout=PIPE, stderr=PIPE)
+        try:
+            _, stderr = deploy_process_execute.communicate(timeout=TIMEOUT)
+        except TimeoutExpired:
+            deploy_process_execute.kill()
+            raise
+        # Deploy should not fail
+        self.assertNotEqual(deploy_process_execute.returncode, 0)
```
Comment (TheSriram) [inline tests/integration/deploy/test_deploy_command.py]: should this fail be a non zero exit code?

File: tests/integration/deploy/test_deploy_command.py [hunk 3]
```diff
@@ -344,6 +344,110 @@ def test_deploy_with_s3_bucket_switch_region(self, template_file):
...
+        deploy_process_execute = Popen(deploy_command_list, stdout=PIPE)
+        try:
+            deploy_process_execute.communicate(timeout=TIMEOUT)
+        except TimeoutExpired:
+            deploy_process_execute.kill()
+            raise
+        # Deploy should succeed
+        self.assertEqual(deploy_process_execute.returncode, 0)
+
+        # Try to deploy to another region.
```
Comment (TheSriram) [inline tests/integration/deploy/test_deploy_command.py]: same here about the region.

File: tests/integration/deploy/test_deploy_command.py (hunk 1 above)
Comment (jfuss) [reply tests/integration/deploy/test_deploy_command.py]: updated

File: tests/integration/deploy/test_deploy_command.py (hunk 3 above)
Comment (jfuss) [reply tests/integration/deploy/test_deploy_command.py]: Updated

File: tests/integration/deploy/test_deploy_command.py (hunk 2 above)
Comment (jfuss) [reply tests/integration/deploy/test_deploy_command.py]: Yes. This is testing that if you deploy the same things twice and add `--fail-on-empty-changeset` the command fails.

Comment (TheSriram) [reply tests/integration/deploy/test_deploy_command.py]: oops, sorry, I thought you were using assertEqual not assertNotEqual

**Pull request merged**